app/src/main/java/com/example/tasky/feature_agenda/presentation/event_detail_screen/EventDetailViewModel.kt

```diff
@@ -118,23 +151,144 @@ class EventDetailViewModel @Inject constructor(
                     }
                 }
             }
+
+            is EventDetailOnClick.DateTimePickerChanged -> {
+                _state.update {
+                    it.copy(dateTimePicker = event.dateTimeOption)
+                }
+            }
+
+            is EventDetailOnClick.FromDateChanged -> {
+                _state.update {
+                    it.copy(fromDate = event.fromDate)
+                }
+            }
+            is EventDetailOnClick.FromTimeChanged -> {
+                _state.update {
+                    it.copy(fromTime = event.fromTime)
+                }
+            }
```

Comment:
If this gets set to a time that is after `toTime`, I'd also adjust the `toTime` here to something like `fromTime + 30min` to avoid having invalid times. Same counts for the dates of course